### PR TITLE
add inline patch document

### PR DESCRIPTION
**Pull request by Liujingfang1**
We'll cut a release after this doc is merged

### CODE REVIEW

File: examples/inlinePatch.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+Create an overlay and add an inline patch in `patchesStrategicMerge` field to the kustomization file
+to change the image from `nignx` to `nignx:latest`.
+
+<!-- @addSMPatch @test -->
+```
+SMP_OVERLAY=$DEMO_HOME/smp
+mkdir $SMP_OVERLAY
+cat <<EOF >$SMP_OVERLAY/kustomization.yaml
+bases:
```
Comment (richardmarshall) [inline examples/inlinePatch.md]: Would be better to use `resources:` here instead.

File: examples/inlinePatch.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+Create an overlay and add an inline patch in `patchesJSON6902` field to the kustomization file
+to change the image from `nignx` to `nignx:latest`.
+
+<!-- @addJSONPatch @test -->
+```
+JSON_OVERLAY=$DEMO_HOME/json
+mkdir $JSON_OVERLAY
+cat <<EOF >$JSON_OVERLAY/kustomization.yaml
+bases:
```
Comment (richardmarshall) [inline examples/inlinePatch.md]: Same here.

File: examples/inlinePatch.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+Create an overlay and add an inline patch in `patches` field to the kustomization file
+to change the image from `nignx` to `nignx:latest`.
+
+<!-- @addPatch @test -->
+```
+PATCH_OVERLAY=$DEMO_HOME/patch
+mkdir $PATCH_OVERLAY
+cat <<EOF > $PATCH_OVERLAY/kustomization.yaml
+bases:
```
Comment (richardmarshall) [inline examples/inlinePatch.md]: And here as well.

File: examples/inlinePatch.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          image: nginx:latest
+          args:
+          - one
+          - two
+```
+
+## Inline Patch for PatchesJson6902
+
+Create an overlay and add an inline patch in `patchesJSON6902` field to the kustomization file
+to change the image from `nignx` to `nignx:latest`.
```
Comment (khrisrichardson) [inline examples/inlinePatch.md]: `s/nignx/nginx/g`

File: examples/inlinePatch.md (hunk 1 above)
Comment (Liujingfang1) [reply examples/inlinePatch.md]: done

File: examples/inlinePatch.md (hunk 2 above)
Comment (Liujingfang1) [reply examples/inlinePatch.md]: done

File: examples/inlinePatch.md (hunk 3 above)
Comment (Liujingfang1) [reply examples/inlinePatch.md]: done

File: examples/inlinePatch.md (hunk 4 above)
Comment (Liujingfang1) [reply examples/inlinePatch.md]: done

**Comment by Liujingfang1** [comment]
Also add an example using `$patch: delete` in the Strategic Merge Patch to make sure it works in the inline patch.

**Pull request merged**